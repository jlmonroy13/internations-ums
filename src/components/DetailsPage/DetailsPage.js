import React from 'react';
import PropTypes from 'prop-types';
import SidebarNav from '../SidebarNav/';
import LoadingMessage from '../LoadingMessage/';
import { ListUsersOfGroup, ListGroupsOfUser } from '../List/';
import EditMenu from '../EditMenu';
import DetailsInfo from '../DetailsInfo';
import DetailsForm from '../DetailsForm';

function DetailsPage({ isAnUser, type, data, editMode }) {
  if (data === null) return <LoadingMessage />;
  if (!data.id) return <div>Invalid</div>;
  const { firstName, name, user, group, id } = data;
  let itemsListArr = [];
  if (user) itemsListArr = Object.keys(user);
  if (group) itemsListArr = Object.keys(group);
  const listLength = itemsListArr.length;
  return (
    <div className="dashboard__body">
      <SidebarNav />
      <div className="details">
        <div className="grid">
          <div className="grid__item medium--two-fifths one-whole">
            <EditMenu type={type} id={id} listLength={listLength} itemsList={itemsListArr} />
            {editMode ? (
              <DetailsForm data={data} isAnUser={isAnUser} />
            ) : (
              <DetailsInfo data={data} isAnUser={isAnUser} listLength={listLength} />
            )}
          </div>
          <div className="grid__item medium--three-fifths one-whole">
            <div className="relative">
              {isAnUser ? (
                <ListGroupsOfUser page="details" type={type} title={`${firstName}'s Groups`} />
              ) : (
                <ListUsersOfGroup page="details" type={type} title={`${name}'s Users`} />
              )}
            </div>
          </div>
        </div>
      </div>
    </div>
  );
}

DetailsPage.propTypes = {
  type: PropTypes.string.isRequired,
  isAnUser: PropTypes.bool.isRequired,
  data: PropTypes.object,
  editMode: PropTypes.bool,
};

export default DetailsPage;
