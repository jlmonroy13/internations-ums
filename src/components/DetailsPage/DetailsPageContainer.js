import { connect } from 'react-redux';
import { firebaseConnect, dataToJS, isLoaded, isEmpty } from 'react-redux-firebase';
import DetailsPage from './DetailsPage';
import StrUtils from '../../utils/string';

const fbWrappedComponent = firebaseConnect(['/contacts', '/groups'])(DetailsPage);

function mapStateToProps(state) {
  const {
    firebase,
    routing: { locationBeforeTransitions: { pathname } },
    events: { editMode },
  } = state;
  const type = StrUtils.getStringBetween(pathname, '/', '/');
  const id = pathname.slice(type.length + 2);
  const isAnUser = type === 'user';
  const ref = type === 'user' ? `contacts/${id}` : `${type}s/${id}`;
  let data = dataToJS(firebase, ref);
  if (!isLoaded(data)) {
    data = null;
  } else if (isEmpty(data)) {
    data = {};
  }
  return {
    type,
    isAnUser,
    data,
    editMode,
  };
}

export default connect(mapStateToProps, null)(fbWrappedComponent);
