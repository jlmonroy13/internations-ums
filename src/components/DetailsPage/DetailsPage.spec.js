import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import DetailsPage from './DetailsPage';
import { reactReduxFirebase } from 'react-redux-firebase';
import fbConfig from '../../constants/firebaseConfig';
import { createStore, compose, combineReducers } from 'redux';

describe('<DetailsPage />', () => {
  it('should render the details page component', () => {
    const data = {
      firstName: '',
      lastName: '',
      email: '',
      facebook: '',
      twitter: '',
      urlPhoto: '',
      name: '',
      description: '',
      user: '',
      group: '',
      id: '',
    };
    const createStoreWithMiddleware = compose(
      reactReduxFirebase(fbConfig, { userProfile: 'users' }),
      typeof window === 'object' && typeof window.devToolsExtension !== 'undefined'
        ? window.devToolsExtension()
        : f => f,
    )(createStore);
    const store = createStoreWithMiddleware(combineReducers({ test: (state = {}) => state }));
    const wrapper = renderer.create(
      <Provider store={store}>
        <DetailsPage isAnUser={true} type="" data={data} />
      </Provider>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
