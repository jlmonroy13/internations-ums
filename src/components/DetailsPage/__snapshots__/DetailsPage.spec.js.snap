// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DetailsPage /> should render the details page component 1`] = `
<div>
  Invalid
</div>
`;
